### PR TITLE
LoopVectorize: add test for crash in #72969

### PR DESCRIPTION
**Pull request by artagnon**

### CODE REVIEW

File: llvm/test/Transforms/LoopVectorize/pr72969.ll [hunk 1]
```diff
@@ -0,0 +1,26 @@
+; RUN: not --crash opt -passes=loop-vectorize -S < %s
+
+target triple = "x86_64-unknown-linux-gnu"
```
Comment (fhahn) [inline llvm/test/Transforms/LoopVectorize/pr72969.ll:3]: If this depends on the X86 backend, it needs to go in the `X86` subdirectory so it only runs when the X86 backend is built. Can it be made target-independent by using `-force-vector-width`/`-force-vector-interleave`?

Comment (fhahn) [reply llvm/test/Transforms/LoopVectorize/pr72969.ll]: Ah right, I don't think it's necessary to duplicate the test, having it X86 only is fine.

**Comment by david-arm** [comment]
What's the best thing to do with this PR? The test seems fine, but if it should be added as part of the actual fix does it make sense to close this and just review https://github.com/llvm/llvm-project/pull/74456?

**Comment by artagnon** [comment]
Hi @david-arm,

Sorry I'm behind on my PRs: I'm currently on vacation, due to return to work in mid-March.

As for this PR, since the test is fine, I think we can merge it as-is, to document the failure. I'll try to pick up #74456 soon and complete it. What do you think?

**Review by david-arm — APPROVED** [review]
LGTM!

**Comment by david-arm** [comment]
> Hi @david-arm,
> 
> Sorry I'm behind on my PRs: I'm currently on vacation, due to return to work in mid-March.
> 
> As for this PR, since the test is fine, I think we can merge it as-is, to document the failure. I'll try to pick up #74456 soon and complete it. What do you think?

Yeah, seems reasonable. And no rush to finish the other patch - enjoy your vacation!

**Comment by artagnon** [comment]
I got a [buildbot failure](https://lab.llvm.org/buildbot/#/builders/98/builds/33789) post-merge, although the native CI seems to have passed it.

**Comment by nikic** [comment]
And this is why I told you not to pre-commit tests for crashes :) If you do this, you need to at least make it a `REQUIRES: asserts` test, to be reasonably confident that it will always crash.

**Comment by artagnon** [comment]
Thanks for the suggestion; will attempt a quick fix. [I really shouldn't have merged something while on vacation: this is what I get]

**Pull request merged**